packages/react-native/ReactAndroid/src/main/java/com/facebook/react/fabric/mounting/mountitems/IntBufferBatchMountItem.java

```diff
@@ -117,70 +123,226 @@ public void execute(@NonNull MountingManager mountingManager) {
 
     beginMarkers("mountViews");
 
+    ArrayList<IntBufferMountItem> mutationsArray = new ArrayList<>();
+
+    // we put views in the mount item objects also for some operations since tags
+    // are only mapped to ViewState which are internal.
+    // In most cases we will need the view to perform an action in the listener
     int i = 0, j = 0;
     while (i < mIntBufferLen) {
       int rawType = mIntBuffer[i++];
       int type = rawType & ~INSTRUCTION_FLAG_MULTIPLE;
       int numInstructions = ((rawType & INSTRUCTION_FLAG_MULTIPLE) != 0 ? mIntBuffer[i++] : 1);
       for (int k = 0; k < numInstructions; k++) {
+        IntBufferMountItem item;
         if (type == INSTRUCTION_CREATE) {
           String componentName = getFabricComponentName((String) mObjBuffer[j++]);
-          surfaceMountingManager.createView(
-              componentName,
-              mIntBuffer[i++],
-              mObjBuffer[j++],
-              castToState(mObjBuffer[j++]),
-              castToEventEmitter(mObjBuffer[j++]),
-              mIntBuffer[i++] == 1);
+          int reactTag = mIntBuffer[i++];
+          View view = null;
+          if (surfaceMountingManager.getViewExists(reactTag)) {
```

Comment:
Other than creating objects we are increasing cost of this method my resolving Views earlier just to share with the listener, but we are not reusing these views on the call to surfaceMountingManager below, again this method is part of the critical path, and this will bring unnecessary regressions